kvbc/src/ReplicaImp.cpp

```diff
@@ -63,19 +63,17 @@ Status ReplicaImp::start() {
 }
 
 void ReplicaImp::createReplicaAndSyncState() {
-  bool isNewStorage = m_metadataStorage->isNewStorage();
-  bool erasedMetaData;
   m_replicaPtr = bftEngine::IReplica::createNewReplica(
-      replicaConfig_, m_cmdHandler, m_stateTransfer, m_ptrComm, m_metadataStorage, erasedMetaData, pm_);
-  if (erasedMetaData) isNewStorage = true;
-  LOG_INFO(logger, "createReplicaAndSyncState: isNewStorage= " << isNewStorage);
-  if (!replicaConfig_.isReadOnly && !isNewStorage && !m_stateTransfer->isCollectingState()) {
-    uint64_t removedBlocksNum = replicaStateSync_->execute(
-        logger, *m_kvBlockchain, getLastReachableBlockNum(), m_replicaPtr->getLastExecutedSequenceNum());
-    LOG_INFO(logger,
-             "createReplicaAndSyncState: removedBlocksNum = "
-                 << removedBlocksNum << ", new m_lastBlock = " << getLastBlockNum()
-                 << ", new m_lastReachableBlock = " << getLastReachableBlockNum());
+      replicaConfig_, m_cmdHandler, m_stateTransfer, m_ptrComm, m_metadataStorage, pm_);
+  const auto lastExecutedSeqNum = m_replicaPtr->getLastExecutedSequenceNum();
+  LOG_INFO(logger, KVLOG(lastExecutedSeqNum));
+  if (!replicaConfig_.isReadOnly && !m_stateTransfer->isCollectingState()) {
+    try {
+      uint64_t removedBlocksNum = replicaStateSync_->execute(logger, *m_kvBlockchain, lastExecutedSeqNum);
+      LOG_INFO(logger, KVLOG(lastExecutedSeqNum, removedBlocksNum, getLastBlockNum(), getLastReachableBlockNum()));
+    } catch (std::exception &e) {
+      std::terminate();
```

Comment:
Does it make sense to log before calling terminate()?

Reply:
"throw" has an information in it

Reply:
Yes, I see. Will that end up in the log file, though?

Reply:
I'll add the log just to be sure.